src/query/api/v1/httpd/handler.go

```diff
@@ -181,7 +182,9 @@ func (h *Handler) RegisterRoutes() error {
 	// Wrap requests with response time logging as well as panic recovery.
 	var (
 		wrapped = func(n http.Handler) http.Handler {
-			return logging.WithResponseTimeAndPanicErrorLogging(n, h.instrumentOpts)
+			return httputil.CompressionHandler{
```

Comment:
nit: Want to add a comment that this does compression under the hood? for all wrapped routes?

Reply:
Actually also.. do we want want to do this in `applyMiddleware(...)`?  That means that every single route will get it rather than only those using `wrapped(...)` here.

Reply:
Will do. It actually doesn't force compression on for all requests; rather  it only returns compressed responses provided the request has an `Accept-Encoding:gzip` (or `Accept-Encoding:deflate`) header